torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -261,6 +261,8 @@ class TORCH_API ProcessGroupNCCL : public Backend {
 
     // Constructor takes a list of CUDA devices
     WorkNCCL(
```

Comment:
Can you make `WorkNCCL` a friend class of `ProcessGroupNCCL` so that you don't need to pass in these two. cc: @kwen2501 who proposed this idea originally.

Reply:
WorkNCCL does not have a reference to ProcessGroupNCCL, even make ProcessGroupNCCL a friend class of WorkNCCL, WorkNCCL still can not access pgName_ and pgDesc_.